napalm_ios/ios.py

```diff
@@ -1134,129 +1136,225 @@ def bgp_time_conversion(bgp_uptime):
-
-        Currently, no VRF support
-        Not tested with IPv6
-
-        Example output of 'show ip bgp summary' only peer table
-        Neighbor        V    AS MsgRcvd MsgSent   TblVer  InQ OutQ Up/Down State/PfxRcd
-        10.100.1.1      4   200      26      22      199    0    0 00:14:23 23
-        10.200.1.1      4   300      21      51      199    0    0 00:13:40 0
-        192.168.1.2     4   200      19      17        0    0    0 00:00:21 2
-        1.1.1.1         4     1       0       0        0    0    0 never    Active
-        3.3.3.3         4     2       0       0        0    0    0 never    Idle
-        1.1.1.2         4     1      11       9        0    0    0 00:00:13 Idle (Admin)
-        1.1.1.3         4 27506  256642   11327     2527    0    0 1w0d     519
-        1.1.1.4         4 46887 1015641   19982     2527    0    0 1w0d     365
-        192.168.1.237   4 60000    2139    2355 13683280    0    0 1d11h    4 (SE)
-        10.90.1.4       4 65015    2508    2502      170    0    0 1d17h    163
-        172.30.155.20   4   111       0       0        0    0    0 never    Active
-        1.1.1.5         4  6500      54      28        0    0    0 00:00:49 Idle (PfxCt)
-        10.1.4.46       4  3979   95244   98874   267067    0    0 8w5d     254
-        10.1.4.58       4  3979    2715    3045   267067    0    0 1d21h    2
-        10.1.1.85       4 65417 8344303 8343570      235    0    0 1y28w    2
+        Currently no VRF support
+        Tested and working with IPv6
         """
-        cmd_bgp_summary = 'show ip bgp summary'
-        bgp_neighbor_data = {}
+        supported_afi = ['ipv4', 'ipv6']
+        # start output dict structure
+        bgp_neighbor_data = dict()
         bgp_neighbor_data['global'] = {}
-
-        output = self._send_command(cmd_bgp_summary).strip()
-        # Cisco issue where new lines are inserted after neighbor IP
-        output = re.sub(RE_IPADDR_STRIP, r"\1", output)
-        if 'Neighbor' not in output:
-            return {}
-        for line in output.splitlines():
-            if 'router identifier' in line:
-                # BGP router identifier 172.16.1.1, local AS number 100
-                rid_regex = r'^.* router identifier (\d+\.\d+\.\d+\.\d+), local AS number (\d+)'
-                match = re.search(rid_regex, line)
-                router_id = match.group(1)
-                local_as = int(match.group(2))
-                break
-        bgp_neighbor_data['global']['router_id'] = py23_compat.text_type(router_id)
+        # get summary output from device
+        cmd_bgp_all_sum = 'show bgp all summary'
+        summary_output = self._send_command(cmd_bgp_all_sum).strip()
+        # get neighbor output from device
+        neighbor_output = ''
+        for afi in supported_afi:
+            cmd_bgp_neighbor = 'show bgp %s unicast neighbors' % afi
+            neighbor_output += self._send_command(cmd_bgp_neighbor).strip()
+            neighbor_output += "\n"
```

Comment:
I think I needed to have a newline there, and the strip + append was the easiest way to get it to be deterministic. Hazy memory though...